### ISSUE TITLE
[BUG][Web] Can not download selected photos in public shared album (via public share link)

### ISSUE THREAD

**Issue by tienducle** [immich-app/immich#5998]
### The bug

As discussed in this [discord thread](https://discord.com/channels/979116623879368755/1189317771666935838), when browsing a public shared album using a public share link, the download button always downloads all photos of the album and it is not possible to download only the selected photos.

### The OS that Immich Server is running on

Debian

### Version of Immich Server

v1.91.4

### Version of Immich Mobile App

N/A

### Platform with the issue

- [ ] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
N/A
```

### Your .env content

```Shell
N/A
```

### Reproduction steps

```bash
1. Create a new album
2. Add 3 photos to it
3. Create a public share link
4. View album using public share link
5. Select 2 photos
6. Click download

**Expected result**: ZIP file contains the 2 selected photos

**Actual result**: ZIP file contains all 3 photos
```

**Comment by alextran1502** [comment]
Download in individual public shared link works.

The bug here is in public shared link for an album, the top menu doesn't show up when in selection mode

**Comment by tienducle** [comment]
Do you have any ETA on this?

**Comment by RavidYael** [comment]
Hi everyone,
I've noticed this issue and I'm interested in working on it. I'll do my best to figure it out and submit a solution.

**Comment by jrasm91** [comment]
This has already been fixed by #7227.
